sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -145,11 +145,14 @@ case class Not(child: Expression)
   override def sql: String = s"(NOT ${child.sql})"
 }
 
-/**
- * Evaluates to `true` if `values` are returned in `query`'s result set.
- */
-case class InSubquery(values: Seq[Expression], query: ListQuery)
-  extends Predicate with Unevaluable {
+abstract class PredicateSubquery extends BinaryExpression with Predicate with Unevaluable {
```

Comment:
Besides, using `BinaryExpression` here may cause errors when analyze query. It will throw an exception before the rule `ResolveSubquery`. I'll revert this.